R/MsBackendMzR-functions.R

```diff
@@ -21,14 +21,118 @@ MsBackendMzR <- function() {
 .mzR_header <- function(x = character()) {
     if (length(x) != 1)
         stop("'x' should have length 1")
-    suppressPackageStartupMessages(
-        require("mzR", quietly = TRUE, character.only = TRUE)
-    )
+    requireNamespace("mzR", quietly = TRUE)
     msd <- mzR::openMSfile(x)
     on.exit(mzR::close(msd))
     hdr <- mzR::header(msd)
     colnames(hdr)[colnames(hdr) == "seqNum"] <- "scanIndex"
     colnames(hdr)[colnames(hdr) == "precursorScanNum"] <- "precScanNum"
     colnames(hdr)[colnames(hdr) == "precursorMZ"] <- "precursorMz"
+    colnames(hdr)[colnames(hdr) == "retentionTime"] <- "rtime"
     S4Vectors::DataFrame(hdr)
 }
+
+#' Read peaks from a single mzML file.
+#'
+#' @param x `character(1)` with the file to read from.
+#'
+#' @param scanIndex (required) indices of spectra from which the data should be
+#'     retrieved.
+#'
+#' @author Johannes Rainer
+#'
+#' @return list of `matrix`
+#'
+#' @noRd
+.mzR_peaks <- function(x = character(), scanIndex = integer()) {
+    if (length(x) != 1)
+        stop("'x' should have length 1")
+    msd <- mzR::openMSfile(x)
+    on.exit(mzR::close(msd))
+    ## hd_spectra <- mzR::header(msd, max(scanIndex))
+    pks <- mzR::peaks(msd, scanIndex)
+    if (is.matrix(pks))
+        pks <- list(pks)
+    lapply(pks, function(z) {
+        colnames(z) <- c("mz", "intensity")
+        z
+    })
+}
+
+#' Utility function to convert columns in the `x` `DataFrame` that have only
+#' a single element to `Rle`. Also columns specified with parameter `columns`
+#' will be converted (if present).
+#'
+#' @param x `DataFrame`
+#'
+#' @param columns `character` of column names that should be converted to `Rle`
+#'
+#' @return `DataFrame`
+#'
+#' @author Johannes Rainer
+#'
+#' @importClassesFrom S4Vectors Rle
+#'
+#' @importFrom S4Vectors Rle
+#'
+#' @noRd
+.compress_spectra_data <- function(x, columns = c("fromFile")) {
```

Comment:
I am not convinced with calling this *compressing* a column. Wouldn't it be clearer to calls this `.as_rle_spectra_data`. Similarly replace `.uncompress_spectra_data` with `.as_vector_spectra_data`?

I've had the same thought [here](https://github.com/lgatto/Features/blob/0b85d90a47b6e4b2feb95df03b15af3ec7f3f2a0/R/shrink.R#L75) that I initially called compressing a column, but dropped it (not confined that *shrinking* is better though) because compression has a specific meaning.

Reply:
Agree. `.as_rle_` and `.as_vector_` exactly tells what's doing, so I'll use that.

Reply:
Or `.deflate_`, `.inflate` might also be nice - deflating a blown-up `DataFrame` with just *air*